lib/wallet-e2e/src/Cardano/Wallet/Spec/Network/Node/Cli.hs

```diff
@@ -0,0 +1,64 @@
+module Cardano.Wallet.Spec.Network.Node.Cli where
+
+import qualified Data.Aeson as Aeson
+import qualified Data.String as String
+
+import Cardano.Wallet.Spec.Network.Node
+    ( NodeApi, nodeApiSocket )
+import Data.Aeson
+    ( FromJSON, withObject, (.:) )
+import Path
+    ( toFilePath )
+import Prelude hiding
+    ( stderr, stdout )
+import System.Process.Typed
+    ( ExitCode (..), readProcess, shell )
+
+data NodeTip = NodeTip
+    { block :: Natural
+    , epoch :: Natural
+    , era :: Text
+    , hash :: Text
+    , slot :: Natural
+    , slotInEpoch :: Natural
+    , slotsToEpochEnd :: Natural
+    , syncProgress :: Double
+    }
+    deriving stock (Eq, Show, Generic)
+
+instance FromJSON NodeTip where
```

Comment:
Can we avoid the instance and use it as a  function ?

Reply:
Done